File: ITapply/ITapply.Models/SearchObjects/EmployerSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.SearchObjects
{
    public class EmployerSearchObject : BaseSearchObject
    {
        public string CompanyName { get; set; } = string.Empty;
        public string Industry { get; set; } = string.Empty;
        public int? MinYearsInBusiness { get; set; }
        public int? MaxYearsInBusiness { get; set; }
        public int? LocationId { get; set; }
        public string ContactEmail { get; set; } = string.Empty;
        public VerificationStatus? VerificationStatus { get; set; }
        public string Email { get; set; } = string.Empty;
        public bool? IsActive { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/search_objects/employer_search_object.dart
import 'package:itapply_mobile/models/enums.dart';
import 'package:itapply_mobile/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'employer_search_object.g.dart';

@JsonSerializable()
class EmployerSearchObject extends BaseSearchObject {
  String? CompanyName;
  String? Industry;
  int? MinYearsInBusiness;
  int? MaxYearsInBusiness;
  int? LocationId;
  String? ContactEmail;
  @JsonKey(name: 'VerificationStatus',toJson: _verificationStatusToJson)
  VerificationStatus? verificationStatus;
  String? Email;
  bool? IsActive;

  EmployerSearchObject({
    this.CompanyName,
    this.Industry,
    this.MinYearsInBusiness,
    this.MaxYearsInBusiness,
    this.LocationId,
    this.ContactEmail,
    this.verificationStatus,
    this.Email,
    this.IsActive,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$EmployerSearchObjectToJson(this);
}

int? _verificationStatusToJson(VerificationStatus? status) => status?.index;
